### PR TITLE
LCAM-273-CC-Schemas

### PR DESCRIPTION
**Pull request by loganathanMOJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-273)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsRequest.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      "type": "object",
+      "description": "Crown court overview",
+      "$ref": "common/apiCrownCourtOverview.json"
+    },
+    "IOJAppeal": {
+      "type": "object",
+      "description": "IOJ Appeal",
+      "$ref": "common/apiIOJAppeal.json"
+    },
+    "FinancialAssessment": {
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsRequest.json:59]: Can we make this lower case like the other fields? Same with IOJAppeal

File: crown-court-proceeding/src/main/resources/schemas/common/apiFinancialAssessment.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "id": "apiFinancialAssessment.json",
+  "type": "object",
+  "description": "Crown Court Overview",
+  "properties": {
+    "initResult": {
+      "type": "string",
+      "description": "Outcome of initial assessment"
+    },
+    "fassInitStatus": {
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/resources/schemas/common/apiFinancialAssessment.json:11]: Can we just call this `initStatus`?

File: crown-court-proceeding/src/main/resources/schemas/common/apiFinancialAssessment.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      "description": "Assessment Status",
+      "existingJavaType": "uk.gov.justice.laa.crime.crowncourt.staticdata.enums.CurrentStatus"
+    },
+    "fullResult": {
+      "type": "string",
+      "description": "Outcome of full assessment"
+    },
+    "fassFullStatus": {
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/resources/schemas/common/apiFinancialAssessment.json:19]: Can we just call this `fullStatus`?

File: crown-court-proceeding/src/main/resources/schemas/common/apiFinancialAssessment.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      "existingJavaType": "uk.gov.justice.laa.crime.crowncourt.staticdata.enums.CurrentStatus"
+    },
+    "hardshipOverview": {
+      "type": "object",
+      "description": "Hardship over iew",
+      "$ref": "apiHardshipOverview.json"
+    }
+  },
+  "additionalProperties": false,
+  "required": ["initResult", "fassInitStatus", "fullResult", "fassFullStatus"]
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/resources/schemas/common/apiFinancialAssessment.json:30]: It's possible there might not be a `fullResult` and `fassFullStatus` if it's an initial assessment

File: crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsResponse.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      "title": "Rep Order decision"
+    },
+    "ccRepType": {
+      "type": "string",
+      "title": "crown court rep type"
+    },
+    "ccRepOrderDate": {
+      "type": "string",
+      "title": "crown court withdrawal date",
+      "format": "date-time"
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsResponse.json:19]: Can we simplify the names of the fields? It seems redundant to prepend each with cc when they're in the Crown Court Response object

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/staticdata/enums/MagCourtOutcome.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    APPEAL_TO_CC("APPEAL TO CC", "Appeal to Crown Court");
+
+    private String outcome;
+    private String description;
+
+    public static MagCourtOutcome getFrom(String outcome) {
+        if (StringUtils.isBlank(outcome)) return null;
+
+        return Stream.of(MagCourtOutcome.values())
+                .filter(f -> f.outcome.equals(outcome))
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/staticdata/enums/MagCourtOutcome.java]: Fixed now

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/staticdata/enums/CurrentStatus.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.validation.constraints.NotNull;
+import java.util.stream.Stream;
+
+/**
+ * static data migrated from TOGDATA.HARDSHIP_REVIEW_STATUSES,
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/staticdata/enums/CurrentStatus.java]: Thanks. Fixed now

File: crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsRequest.json [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    "magCourtOutcome": {
+      "type": "object",
+      "description": "Magistrate Court Outcome",
+      "existingJavaType": "uk.gov.justice.laa.crime.crowncourt.staticdata.enums.MagCourtOutcome"
+    },
+    "iojResult": {
+      "type": "string",
+      "description": "Ioj Result"
+    },
+    "decisionReason": {
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsRequest.json:26]: Thanks. Fixed now

File: crown-court-proceeding/src/main/resources/schemas/common/apiCrownCourtOverview.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "id": "apiCrownCourtOverview.json",
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crown-court-proceeding/src/main/resources/schemas/common/apiCrownCourtOverview.json]: Removed apiCrownCourtOverview.json from schemas

File: crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsRequest.json [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+      "type": "string",
+      "description": "Decision Date",
+      "format": "date-time"
+    },
+    "committalDate": {
+      "type": "string",
+      "description": "Committal Date",
+      "format": "date-time"
+    },
+    "dateReceived": {
```
[the remark this answers is not in the document]
Comment (loganathanMOJ) [reply crown-court-proceeding/src/main/resources/schemas/apiCheckCrownCourtActionsRequest.json:41]: We need this filed for "determine_crown_reporder_date" stored procedure

**Review by Mhowell494 — APPROVED** [review]

**Pull request merged**